From dea387c18f88151f62eb4e9536c3a71aa2163f9d Mon Sep 17 00:00:00 2001
From: dylandoesprogramming664 <dylan20xx@outlook.com>
Date: Thu, 22 Jan 2026 09:53:01 -0500
Subject: [PATCH 4/6] No 1/256 glitch; fast text speed by default.

Gameplay Changes:
- No 1/256 glitch, regardless of difficulty

Miscellaneous Changes:
- Text speed is now TEXT_DELAY_FAST by default

---
 engine/battle/core.asm     | 14 --------------
 engine/menus/main_menu.asm |  1 -
 2 files changed, 15 deletions(-)

diff --git a/engine/battle/core.asm b/engine/battle/core.asm
index 1535c82..1528d22 100644
--- a/engine/battle/core.asm
+++ b/engine/battle/core.asm
@@ -4856,19 +4856,9 @@ CriticalHitTest:
 	jr nc, .noFocusEnergyUsed
 	ld b, $ff                    ; cap at 255/256
 .noFocusEnergyUsed
-	ld a, [wDifficulty] ; Check if player is on hard mode
-	and a
-	jr nz, .NotGuarenteedCrit ; keep 1/256 chance to not crit if on hard mode
 	ld a, b
 	inc a ; optimization of "cp $ff"
 	jr z, .guaranteedCriticalHit
-.NotGuarenteedCrit
-	call BattleRandom            ; generates a random value, in "a"
-	rlc a
-	rlc a
-	rlc a
-	cp b                         ; check a against calculated crit rate
-	ret nc                       ; no critical hit if no borrow
 .guaranteedCriticalHit
 	ld a, $1
 	ld [wCriticalHitOrOHKO], a   ; set critical hit flag
@@ -5685,14 +5675,10 @@ MoveHitTest:
 	ld a, [wEnemyMoveAccuracy]
 	ld b, a
 .doAccuracyCheck
-	ld a, [wDifficulty] ; Check if player is on hard mode
-	and a
-	jr nz, .DontRemoveMiss ; Keep 1/256 chance to miss on hard mode
 	; The following snippet fixes 1/256 chance to miss on 100% accurate moves bug on normal mode
 	ld a, b
 	cp $FF ; Is the value $FF?
 	ret z ; If so, we need not calculate, just so we can fix this bug.
-.DontRemoveMiss
 	call BattleRandom
 	cp b
 	jr nc, .moveMissed
diff --git a/engine/menus/main_menu.asm b/engine/menus/main_menu.asm
index 405c528..b5dd8b3 100644
--- a/engine/menus/main_menu.asm
+++ b/engine/menus/main_menu.asm
@@ -125,7 +125,6 @@ MainMenu:
 InitOptions:
 	ld a, TEXT_DELAY_FAST
 	ld [wLetterPrintingDelayFlags], a
-	ld a, TEXT_DELAY_MEDIUM
 	ld [wOptions], a
 	ld a, 64 ; audio?
 	ld [wPrinterSettings], a
-- 
2.52.0

